modules/azure/automationaccount.go

```diff
@@ -0,0 +1,342 @@
+	_, err := GetAutomationAccountE(t, automationAccountName, resourceGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// AutomationAccountDscExists indicates whether the specified Azure Automation Account DSC exists.
+// This function would fail the test if there is an error.
+func AutomationAccountDscExists(t testing.TestingT, dscConfiguraitonName string, automationAccountName string, resourceGroupName string, subscriptionID string) bool {
+	exists, err := AutomationAccountDscExistsE(t, dscConfiguraitonName, automationAccountName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return exists
+}
+
+// AutomationAccountDscExistsE indicates whether the specified Azure Automation Account exists.
+func AutomationAccountDscExistsE(t testing.TestingT, dscConfiguraitonName string, automationAccountName string, resourceGroupName string, subscriptionID string) (bool, error) {
+	_, err := GetAutomationAccountDscConfigurationE(t, dscConfiguraitonName, automationAccountName, resourceGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// AutomationAccountDscCompiled indicates whether the specified Azure Automation Account DSC compiled successfully.
+// This function would fail the test if there is an error.
+func AutomationAccountDscCompiled(t testing.TestingT, dscConfiguraitonName string, automationAccountName string, resourceGroupName string, subscriptionID string) bool {
+	compiled, err := AutomationAccountDscCompiledE(t, dscConfiguraitonName, automationAccountName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return compiled
+}
+
+// AutomationAccountDscCompiledE indicates whether the specified Azure Automation Account DSC compiled successfully.
+// DSC compilation is performed via a Terraform null_resource using PowerShell Core, executing async.
+// Compilation can take a few minutes to spin up resources requiring a retry mechanism to allow for this
+func AutomationAccountDscCompiledE(t testing.TestingT, dscConfiguraitonName string, automationAccountName string, resourceGroupName string, subscriptionID string) (bool, error) {
+	seconds := 30 // 30 second initial delay
+	sleep := time.Duration(seconds) * time.Second
+	maxAttempts := 5 // try 5 times, doubling the delay each time.
+
+	for i := 1; i < maxAttempts; i++ {
+		time.Sleep(sleep)
+		dscCompileJobStatus, err := AutomationAccountDscCompileJobStatusE(t, dscConfiguraitonName, automationAccountName, resourceGroupName, subscriptionID)
+		if err != nil {
+			return false, err
+		}
+		// check if status === Completed
+		if dscCompileJobStatus == "Completed" {
+			return true, nil
+		}
+		sleep = 2 * sleep
+	}
+	return false, nil
```

Comment:
add code comment under what condition would false be returned

Reply:
Fixed in latest commit.